### ISSUE TITLE
Target [Illuminate\Contracts\Container\Container] is not instantiable while building [LaravelDoctrine\ACL\RegisterMappedEventSubscribers]

### ISSUE THREAD

**Issue by manticorp** [laravel-doctrine/acl#25]
Using PHPUnit 4.8.26 on Laravel 5.2 I get the following on certain routes:

```
Illuminate\Contracts\Container\BindingResolutionException: Target [Illuminate\Contracts\Container\Container] is not instantiable while building [LaravelDoctrine\ACL\RegisterMappedEventSubscribers, LaravelDoctrine\ACL\RegisterMappedEventSubscribers, LaravelDoctrine\ACL\RegisterMappedEventSubscribers, LaravelDoctrine\ACL\RegisterMappedEventSubscribers]
```

Full stack trace:

```
Illuminate\Contracts\Container\BindingResolutionException: Target [Illuminate\Contracts\Container\Container] is not instantiable while building [LaravelDoctrine\ACL\RegisterMappedEventSubscribers, LaravelDoctrine\ACL\RegisterMappedEventSubscribers, LaravelDoctrine\ACL\RegisterMappedEventSubscribers, LaravelDoctrine\ACL\RegisterMappedEventSubscribers].

\vendor\laravel\framework\src\Illuminate\Container\Container.php:748
\vendor\laravel\framework\src\Illuminate\Container\Container.php:629
\vendor\laravel\framework\src\Illuminate\Foundation\Application.php:697
\vendor\laravel\framework\src\Illuminate\Container\Container.php:849
\vendor\laravel\framework\src\Illuminate\Container\Container.php:804
\vendor\laravel\framework\src\Illuminate\Container\Container.php:775
\vendor\laravel\framework\src\Illuminate\Container\Container.php:629
\vendor\laravel\framework\src\Illuminate\Foundation\Application.php:697
\vendor\laravel-doctrine\orm\src\DoctrineManager.php:80
\vendor\laravel-doctrine\orm\src\DoctrineManager.php:61
\vendor\laravel-doctrine\orm\src\DoctrineManager.php:39
\vendor\laravel-doctrine\orm\src\BootChain.php:29
\vendor\laravel-doctrine\orm\src\DoctrineServiceProvider.php:128
\vendor\laravel\framework\src\Illuminate\Container\Container.php:1031
\vendor\laravel\framework\src\Illuminate\Container\Container.php:996
\vendor\laravel\framework\src\Illuminate\Container\Container.php:648
\vendor\laravel\framework\src\Illuminate\Foundation\Application.php:697
\vendor\laravel-doctrine\orm\src\DoctrineServiceProvider.php:98
\vendor\laravel\framework\src\Illuminate\Container\Container.php:731
\vendor\laravel\framework\src\Illuminate\Container\Container.php:629
\vendor\laravel\framework\src\Illuminate\Foundation\Application.php:697
\vendor\laravel\framework\src\Illuminate\Container\Container.php:1174
\vendor\laravel\framework\src\Illuminate\Support\Facades\Facade.php:151
\vendor\laravel\framework\src\Illuminate\Support\Facades\Facade.php:120
\vendor\laravel\framework\src\Illuminate\Support\Facades\Facade.php:108
\vendor\laravel\framework\src\Illuminate\Support\Facades\Facade.php:84
\vendor\laravel\framework\src\Illuminate\Support\Facades\Facade.php:65
\vendor\laravel\framework\src\Illuminate\Support\Facades\Facade.php:51
```

I've tried a few things that came to mind, but I was wondering if you have any idea what might be causing this?

**Comment by lokipl** [comment]
I have the same issue. Did you find a solution?

**Comment by manticorp** [comment]
@lokipl Afraid not... :-1:

**Comment by FoxxMD** [comment]
In what order do you have laravel-doctrine service providers?

**Comment by patrickbrouwers** [comment]
Seems like the Container Contract is not bound in your tests. 

Try:
`$this->alias('app', Illuminate\Contracts\Container\Container::class);`

**Comment by manticorp** [comment]
@FoxxMD 

The order is:

```
        /**
         * Doctrine
         */
        LaravelDoctrine\ORM\DoctrineServiceProvider::class,
        LaravelDoctrine\Extensions\GedmoExtensionsServiceProvider::class,
        LaravelDoctrine\ACL\AclServiceProvider::class,
        LaravelDoctrine\Migrations\MigrationsServiceProvider::class,
        LaravelDoctrine\ORM\Auth\Passwords\PasswordResetServiceProvider::class,
```

@patrickbrouwers 

Not quite sure where I'm supposed to add this? I've tried in my AppServiceProvider, no difference, and in tests, no difference...

**Comment by lokipl** [comment]
@patrickbrouwers 
nothing has changed

I'm also tried different order of providers with no progress.

**Comment by patrickbrouwers** [comment]
I'm planning on doing some open source stuff this evening, will see if I can reproduce this myself.

**Comment by lokipl** [comment]
@patrickbrouwers Did you find a solution?

**Comment by patrickbrouwers** [comment]
No sorry not yet. Have been too busy

**Comment by msuarezecca** [comment]
Hello!

I followed this [link](https://laracasts.com/discuss/channels/laravel/phpunit-bindingresolutionexception), and the test worked...

Is not the solution but maybe it's a clue.

It seems to be a problem when the app is destroyed for each test.

Hope it helps...

**Comment by msuarezecca** [comment]
My setup is Laravel 5.2 with PHP 7 and phpunit 5.4

Ping to @lokipl  and @manticorp if they want to give a try...

**Comment by manticorp** [comment]
@msuarezecca Unfortunately, I'd seen this link before - commenting out

```
$this->app->flush();
$this->app = null;
```

unfortunately breaks a lot of my other tests :(

**Comment by manticorp** [comment]
I just tried upgrading to Laravel 5.4 and am now getting the same error but on every page of the site, not just while testing:

```
Target [Doctrine\Common\Annotations\Reader] is not instantiable while building [App\Http\Controllers\XYZController, LaravelDoctrine\ACL\Mappings\Subscribers\BelongsToOrganisationsSubscriber]
```

**Comment by dave-redfern** [comment]
@manticorp Seems to be the ACL service provider.

**Comment by dave-redfern** [comment]
@manticorp It's the registration of the mapped event subscribers, it is still passing arguments to ->container->make() but that was deprecated with 5.4 so it's failing.

**Comment by dave-redfern** [comment]
@patrickbrouwers Hacky solution for ACL service provider is something like changing ln 49 in RegisterMappedEventSubscribers to:

`new $subscriber($configuration->getMetadataDriverImpl()->getReader(), $this->container['config'])`

Or can define an annotation reader concrete that can be loaded by the subscribers.

**Comment by aftabnaveed** [comment]
Has this issue been resolved now? I am still getting the same error.

**Comment by dave-redfern** [comment]
@aftabnaveed It was fixed in 1.0.16. Have you pinned to a specific version? Try updating your composer.json with 1.0.16 for the acl component.

**Comment by cooperaj** [comment]
I'm running 1.0.16 and am seeing this issue under PHPUnit 5.7.19 and Laravel 5.2.45.

<details>
<summary>Click for stacktrace</summary>
<pre>
/build/vendor/laravel/framework/src/Illuminate/Foundation/Testing/Concerns/InteractsWithPages.php:196
/build/vendor/laravel/framework/src/Illuminate/Foundation/Testing/Concerns/InteractsWithPages.php:80
/build/vendor/laravel/framework/src/Illuminate/Foundation/Testing/Concerns/InteractsWithPages.php:61
/build/tests/functional/authentication/CanLoginTest.php:10

Caused by
Illuminate\Contracts\Container\BindingResolutionException: Target [Illuminate\Contracts\Container\Container] is not instantiable while building [LaravelDoctrine\ACL\RegisterMappedEventSubscribers]. in /build/vendor/laravel/framework/src/Illuminate/Container/Container.php:748
Stack trace:
#0 /build/vendor/laravel/framework/src/Illuminate/Container/Container.php(629): Illuminate\Container\Container->build('Illuminate\\Cont...', Array)
#1 /build/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(697): Illuminate\Container\Container->make('Illuminate\\Cont...', Array)
#2 /build/vendor/laravel/framework/src/Illuminate/Container/Container.php(849): Illuminate\Foundation\Application->make('Illuminate\\Cont...')
#3 /build/vendor/laravel/framework/src/Illuminate/Container/Container.php(804): Illuminate\Container\Container->resolveClass(Object(ReflectionParameter))
#4 /build/vendor/laravel/framework/src/Illuminate/Container/Container.php(773): Illuminate\Container\Container->getDependencies(Array, Array)
#5 /build/vendor/laravel/framework/src/Illuminate/Container/Container.php(629): Illuminate\Container\Container->build('LaravelDoctrine...', Array)
#6 /build/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(697): Illuminate\Container\Container->make('LaravelDoctrine...', Array)
#7 /build/vendor/laravel-doctrine/orm/src/DoctrineManager.php(80): Illuminate\Foundation\Application->make('LaravelDoctrine...')
#8 /build/vendor/laravel-doctrine/orm/src/DoctrineManager.php(61): LaravelDoctrine\ORM\DoctrineManager->callExtendOn('default', 'LaravelDoctrine...', Object(LaravelDoctrine\ORM\IlluminateRegistry))
#9 [internal function]: LaravelDoctrine\ORM\DoctrineManager->LaravelDoctrine\ORM\{closure}(Object(LaravelDoctrine\ORM\IlluminateRegistry), Object(LaravelDoctrine\ORM\DoctrineManager))
#10 /build/vendor/laravel-doctrine/orm/src/DoctrineManager.php(39): call_user_func_array(Object(Closure), Array)
#11 [internal function]: LaravelDoctrine\ORM\DoctrineManager->LaravelDoctrine\ORM\{closure}(Object(LaravelDoctrine\ORM\IlluminateRegistry))
#12 /build/vendor/laravel-doctrine/orm/src/BootChain.php(29): call_user_func(Object(Closure), Object(LaravelDoctrine\ORM\IlluminateRegistry))
#13 /build/vendor/laravel-doctrine/orm/src/DoctrineServiceProvider.php(134): LaravelDoctrine\ORM\BootChain::boot(Object(LaravelDoctrine\ORM\IlluminateRegistry))
#14 /build/vendor/laravel/framework/src/Illuminate/Container/Container.php(1031): LaravelDoctrine\ORM\DoctrineServiceProvider->LaravelDoctrine\ORM\{closure}(Object(LaravelDoctrine\ORM\IlluminateRegistry), Object(Illuminate\Foundation\Application))
#15 /build/vendor/laravel/framework/src/Illuminate/Container/Container.php(994): Illuminate\Container\Container->fireCallbackArray(Object(LaravelDoctrine\ORM\IlluminateRegistry), Array)
#16 /build/vendor/laravel/framework/src/Illuminate/Container/Container.php(648): Illuminate\Container\Container->fireResolvingCallbacks('registry', Object(LaravelDoctrine\ORM\IlluminateRegistry))
#17 /build/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(697): Illuminate\Container\Container->make('registry', Array)
#18 /build/vendor/laravel-doctrine/orm/src/DoctrineServiceProvider.php(107): Illuminate\Foundation\Application->make('registry')
#19 /build/vendor/laravel/framework/src/Illuminate/Container/Container.php(731): LaravelDoctrine\ORM\DoctrineServiceProvider->LaravelDoctrine\ORM\{closure}(Object(Illuminate\Foundation\Application), Array)
#20 /build/vendor/laravel/framework/src/Illuminate/Container/Container.php(629): Illuminate\Container\Container->build(Object(Closure), Array)
#21 /build/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(697): Illuminate\Container\Container->make('em', Array)
#22 /build/vendor/laravel/framework/src/Illuminate/Container/Container.php(1178): Illuminate\Foundation\Application->make('em')
#23 /build/app/Providers/CalendarServiceProvider.php(41): Illuminate\Container\Container->offsetGet('em')
#24 /build/vendor/laravel/framework/src/Illuminate/Container/Container.php(731): App\Providers\CalendarServiceProvider->App\Providers\{closure}(Object(Illuminate\Foundation\Application), Array)
#25 /build/vendor/laravel/framework/src/Illuminate/Container/Container.php(629): Illuminate\Container\Container->build(Object(Closure), Array)
#26 /build/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(697): Illuminate\Container\Container->make('App\\UoN\\Service...', Array)
#27 /build/vendor/laravel/framework/src/Illuminate/Container/Container.php(849): Illuminate\Foundation\Application->make('App\\UoN\\Service...')
#28 /build/vendor/laravel/framework/src/Illuminate/Container/Container.php(804): Illuminate\Container\Container->resolveClass(Object(ReflectionParameter))
#29 /build/vendor/laravel/framework/src/Illuminate/Container/Container.php(773): Illuminate\Container\Container->getDependencies(Array, Array)
#30 /build/vendor/laravel/framework/src/Illuminate/Container/Container.php(629): Illuminate\Container\Container->build('App\\Http\\Contro...', Array)
#31 /build/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(697): Illuminate\Container\Container->make('App\\Http\\Contro...', Array)
#32 /build/vendor/laravel/framework/src/Illuminate/Routing/ControllerDispatcher.php(67): Illuminate\Foundation\Application->make('App\\Http\\Contro...')
#33 /build/vendor/laravel/framework/src/Illuminate/Routing/ControllerDispatcher.php(52): Illuminate\Routing\ControllerDispatcher->makeController('App\\Http\\Contro...')
#34 /build/vendor/laravel/framework/src/Illuminate/Routing/Route.php(174): Illuminate\Routing\ControllerDispatcher->dispatch(Object(Illuminate\Routing\Route), Object(Illuminate\Http\Request), 'App\\Http\\Contro...', 'index')
#35 /build/vendor/laravel/framework/src/Illuminate/Routing/Route.php(140): Illuminate\Routing\Route->runController(Object(Illuminate\Http\Request))
#36 /build/vendor/laravel/framework/src/Illuminate/Routing/Router.php(724): Illuminate\Routing\Route->run(Object(Illuminate\Http\Request))
#37 [internal function]: Illuminate\Routing\Router->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request))
#38 /build/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(52): call_user_func(Object(Closure), Object(Illuminate\Http\Request))
#39 /build/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/VerifyCsrfToken.php(64): Illuminate\Routing\Pipeline->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request))
#40 [internal function]: Illuminate\Foundation\Http\Middleware\VerifyCsrfToken->handle(Object(Illuminate\Http\Request), Object(Closure))
#41 /build/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(136): call_user_func_array(Array, Array)
#42 [internal function]: Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#43 /build/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(32): call_user_func(Object(Closure), Object(Illuminate\Http\Request))
#44 /build/vendor/laravel/framework/src/Illuminate/View/Middleware/ShareErrorsFromSession.php(49): Illuminate\Routing\Pipeline->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request))
#45 [internal function]: Illuminate\View\Middleware\ShareErrorsFromSession->handle(Object(Illuminate\Http\Request), Object(Closure))
#46 /build/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(136): call_user_func_array(Array, Array)
#47 [internal function]: Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#48 /build/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(32): call_user_func(Object(Closure), Object(Illuminate\Http\Request))
#49 /build/vendor/laravel/framework/src/Illuminate/Session/Middleware/StartSession.php(64): Illuminate\Routing\Pipeline->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request))
#50 [internal function]: Illuminate\Session\Middleware\StartSession->handle(Object(Illuminate\Http\Request), Object(Closure))
#51 /build/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(136): call_user_func_array(Array, Array)
#52 [internal function]: Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#53 /build/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(32): call_user_func(Object(Closure), Object(Illuminate\Http\Request))
#54 /build/vendor/laravel/framework/src/Illuminate/Cookie/Middleware/AddQueuedCookiesToResponse.php(37): Illuminate\Routing\Pipeline->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request))
#55 [internal function]: Illuminate\Cookie\Middleware\AddQueuedCookiesToResponse->handle(Object(Illuminate\Http\Request), Object(Closure))
#56 /build/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(136): call_user_func_array(Array, Array)
#57 [internal function]: Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#58 /build/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(32): call_user_func(Object(Closure), Object(Illuminate\Http\Request))
#59 /build/vendor/laravel/framework/src/Illuminate/Cookie/Middleware/EncryptCookies.php(59): Illuminate\Routing\Pipeline->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request))
#60 [internal function]: Illuminate\Cookie\Middleware\EncryptCookies->handle(Object(Illuminate\Http\Request), Object(Closure))
#61 /build/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(136): call_user_func_array(Array, Array)
#62 [internal function]: Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#63 /build/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(32): call_user_func(Object(Closure), Object(Illuminate\Http\Request))
#64 /build/app/Http/Middleware/RequireSecureConnection.php(25): Illuminate\Routing\Pipeline->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request))
#65 [internal function]: App\Http\Middleware\RequireSecureConnection->handle(Object(Illuminate\Http\Request), Object(Closure))
#66 /build/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(136): call_user_func_array(Array, Array)
#67 [internal function]: Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#68 /build/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(32): call_user_func(Object(Closure), Object(Illuminate\Http\Request))
#69 [internal function]: Illuminate\Routing\Pipeline->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request))
#70 /build/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(102): call_user_func(Object(Closure), Object(Illuminate\Http\Request))
#71 /build/vendor/laravel/framework/src/Illuminate/Routing/Router.php(726): Illuminate\Pipeline\Pipeline->then(Object(Closure))
#72 /build/vendor/laravel/framework/src/Illuminate/Routing/Router.php(699): Illuminate\Routing\Router->runRouteWithinStack(Object(Illuminate\Routing\Route), Object(Illuminate\Http\Request))
#73 /build/vendor/laravel/framework/src/Illuminate/Routing/Router.php(675): Illuminate\Routing\Router->dispatchToRoute(Object(Illuminate\Http\Request))
#74 /build/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(246): Illuminate\Routing\Router->dispatch(Object(Illuminate\Http\Request))
#75 [internal function]: Illuminate\Foundation\Http\Kernel->Illuminate\Foundation\Http\{closure}(Object(Illuminate\Http\Request))
#76 /build/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(52): call_user_func(Object(Closure), Object(Illuminate\Http\Request))
#77 /build/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/CheckForMaintenanceMode.php(44): Illuminate\Routing\Pipeline->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request))
#78 [internal function]: Illuminate\Foundation\Http\Middleware\CheckForMaintenanceMode->handle(Object(Illuminate\Http\Request), Object(Closure))
#79 /build/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(136): call_user_func_array(Array, Array)
#80 [internal function]: Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#81 /build/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(32): call_user_func(Object(Closure), Object(Illuminate\Http\Request))
#82 [internal function]: Illuminate\Routing\Pipeline->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request))
#83 /build/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(102): call_user_func(Object(Closure), Object(Illuminate\Http\Request))
#84 /build/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(132): Illuminate\Pipeline\Pipeline->then(Object(Closure))
#85 /build/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(99): Illuminate\Foundation\Http\Kernel->sendRequestThroughRouter(Object(Illuminate\Http\Request))
#86 /build/vendor/laravel/framework/src/Illuminate/Foundation/Testing/Concerns/MakesHttpRequests.php(515): Illuminate\Foundation\Http\Kernel->handle(Object(Illuminate\Http\Request))
#87 /build/vendor/laravel/framework/src/Illuminate/Foundation/Testing/Concerns/InteractsWithPages.php(78): Illuminate\Foundation\Testing\TestCase->call('GET', 'http://test.no...', Array, Array, Array)
#88 /build/vendor/laravel/framework/src/Illuminate/Foundation/Testing/Concerns/InteractsWithPages.php(61): Illuminate\Foundation\Testing\TestCase->makeRequest('GET', 'http://test.no...')
#89 /build/tests/functional/authentication/CanLoginTest.php(10): Illuminate\Foundation\Testing\TestCase->visit('/')
#90 [internal function]: CanLoginTest->iShouldBeRedirectedToTheLoginPage()
#91 /build/vendor/phpunit/phpunit/src/Framework/TestCase.php(1054): ReflectionMethod->invokeArgs(Object(CanLoginTest), Array)
#92 /build/vendor/phpunit/phpunit/src/Framework/TestCase.php(905): PHPUnit_Framework_TestCase->runTest()
#93 /build/vendor/phpunit/phpunit/src/Framework/TestResult.php(686): PHPUnit_Framework_TestCase->runBare()
#94 /build/vendor/phpunit/phpunit/src/Framework/TestCase.php(860): PHPUnit_Framework_TestResult->run(Object(CanLoginTest))
#95 /build/vendor/phpunit/phpunit/src/Framework/TestSuite.php(722): PHPUnit_Framework_TestCase->run(Object(PHPUnit_Framework_TestResult))
#96 /build/vendor/phpunit/phpunit/src/Framework/TestSuite.php(722): PHPUnit_Framework_TestSuite->run(Object(PHPUnit_Framework_TestResult))
#97 /build/vendor/phpunit/phpunit/src/TextUI/TestRunner.php(517): PHPUnit_Framework_TestSuite->run(Object(PHPUnit_Framework_TestResult))
#98 /build/vendor/phpunit/phpunit/src/TextUI/Command.php(186): PHPUnit_TextUI_TestRunner->doRun(Object(PHPUnit_Framework_TestSuite), Array, true)
#99 /build/vendor/phpunit/phpunit/src/TextUI/Command.php(116): PHPUnit_TextUI_Command->run(Array, true)
#100 /build/vendor/phpunit/phpunit/phpunit(52): PHPUnit_TextUI_Command::main()
#101 {main}
</pre>
</details>

**Comment by tempestwf** [comment]
Hi

Was this issue resolved? I am having it as well. Oddly, my first test case does not trigger the error, but my second one (with the same syntax in it: $em = \App::make($this->getEntityManagerClass());) does trigger it. I also tried just app('em'); as well as using the facade.

Here are my composer requires:

```
"require": {
    "php": ">=7.1.0",
    "laravel/framework": "5.3.*",
    "laravel-doctrine/orm": "1.2.*",
    "dingo/api": "1.0.x@dev",
    "tymon/jwt-auth": "0.5.*",
    "laravel-doctrine/extensions": "1.0.*",
    "gedmo/doctrine-extensions": "^2.4",
    "beberlei/DoctrineExtensions": "^1.0",
    "laravel-doctrine/migrations": "1.0.*",
    "laravel-doctrine/acl": "1.0.*"
  },
  "require-dev": {
    "fzaninotto/faker": "~1.4",
    "mockery/mockery": "0.9.*",
    "phpunit/phpunit": "~4.0",
    "symfony/css-selector": "2.8.*|3.0.*",
    "symfony/dom-crawler": "2.8.*|3.0.*"
  },
```

Here is my stack trace:

```
Illuminate\Contracts\Container\BindingResolutionException: Target [Illuminate\Contracts\Container\Container] is not instantiable while building [LaravelDoctrine\ACL\RegisterMappedEventSubscribers].

/home/vagrant/Code/laravel-versioned-api/vendor/laravel/framework/src/Illuminate/Container/Container.php:763
/home/vagrant/Code/laravel-versioned-api/vendor/laravel/framework/src/Illuminate/Container/Container.php:644
/home/vagrant/Code/laravel-versioned-api/vendor/laravel/framework/src/Illuminate/Foundation/Application.php:709
/home/vagrant/Code/laravel-versioned-api/vendor/laravel/framework/src/Illuminate/Container/Container.php:864
/home/vagrant/Code/laravel-versioned-api/vendor/laravel/framework/src/Illuminate/Container/Container.php:819
/home/vagrant/Code/laravel-versioned-api/vendor/laravel/framework/src/Illuminate/Container/Container.php:788
/home/vagrant/Code/laravel-versioned-api/vendor/laravel/framework/src/Illuminate/Container/Container.php:644
/home/vagrant/Code/laravel-versioned-api/vendor/laravel/framework/src/Illuminate/Foundation/Application.php:709
/home/vagrant/Code/laravel-versioned-api/vendor/laravel-doctrine/orm/src/DoctrineManager.php:80
/home/vagrant/Code/laravel-versioned-api/vendor/laravel-doctrine/orm/src/DoctrineManager.php:61
/home/vagrant/Code/laravel-versioned-api/vendor/laravel-doctrine/orm/src/DoctrineManager.php:39
/home/vagrant/Code/laravel-versioned-api/vendor/laravel-doctrine/orm/src/BootChain.php:29
/home/vagrant/Code/laravel-versioned-api/vendor/laravel-doctrine/orm/src/DoctrineServiceProvider.php:134
/home/vagrant/Code/laravel-versioned-api/vendor/laravel/framework/src/Illuminate/Container/Container.php:1046
/home/vagrant/Code/laravel-versioned-api/vendor/laravel/framework/src/Illuminate/Container/Container.php:1009
/home/vagrant/Code/laravel-versioned-api/vendor/laravel/framework/src/Illuminate/Container/Container.php:663
/home/vagrant/Code/laravel-versioned-api/vendor/laravel/framework/src/Illuminate/Foundation/Application.php:709
/home/vagrant/Code/laravel-versioned-api/vendor/laravel-doctrine/orm/src/DoctrineServiceProvider.php:107
/home/vagrant/Code/laravel-versioned-api/vendor/laravel/framework/src/Illuminate/Container/Container.php:746
/home/vagrant/Code/laravel-versioned-api/vendor/laravel/framework/src/Illuminate/Container/Container.php:644
/home/vagrant/Code/laravel-versioned-api/vendor/laravel/framework/src/Illuminate/Foundation/Application.php:709
/home/vagrant/Code/laravel-versioned-api/vendor/laravel/framework/src/Illuminate/Support/Facades/Facade.php:237
/home/vagrant/Code/laravel-versioned-api/packages/tempesttools/common/src/Doctrine/Utility/MakeEmTrait.php:29
/home/vagrant/Code/laravel-versioned-api/packages/tempesttools/common/src/Doctrine/Utility/MakeEmTrait.php:21
/home/vagrant/Code/laravel-versioned-api/tests/CrudCreateTest.php:33
```


Any advice would be greatly apperciated.

Will

**Comment by bigperson** [comment]
I have the same issue.
Laravel 5.5
laravel-doctrine 1.4
laravel-doctrine-acl 1.0.16

**Comment by patrickbrouwers** [comment]
If anybody wants to take over this PR: https://github.com/laravel-doctrine/acl/pull/39 feel free.

**Comment by fredpaist** [comment]
I have the same issue.
Laravel 5.6
laravel-doctrine 1.4
laravel-doctrine-acl 1.0.16
PR #39 would fix it for me, so why not accept it?

**Comment by t-hebert73** [comment]
Is there any update to this? I also am having this issue.

**Comment by rgwsk65** [comment]
Me too, i.e. when using `$this->actingAs($user)`

**Comment by jwillz21** [comment]
having this same issue aswell :/

**Comment by eigan** [comment]
Fix by @mxkh in #42 have been merged. Released in 1.0.18.
